packages/shared/src/components/sources/SourceCard.tsx

```diff
@@ -0,0 +1,173 @@
+    public: true,
+    image,
+    handle,
+    memberInviteRole,
+    memberPostingRole,
+  };
+
+  return (
+    <Card
+      className={classNames(
+        'overflow-hidden p-0',
+        borderColor
+          ? `border-theme-color-${borderColor} hover:border-theme-color-${borderColor}`
+          : 'border-theme-color-avocado hover:border-theme-color-avocado',
+      )}
+    >
+      <div className="h-24 rounded-t-2xl bg-theme-bg-onion">
+        <Image
+          className="object-cover w-full h-full"
+          src={banner || SquadDirectoryCardBannerDefault}
+          alt="Banner image for source, showing abstract colors."
+        />
+      </div>
+      <div className="flex flex-col flex-1 p-4 -mt-12 rounded-t-2xl bg-theme-bg-secondary">
+        <div className="flex justify-between items-end mb-3">
+          {!!image && (
+            <img
+              className="-mt-14 w-24 h-24 rounded-full z-10"
+              src={image}
+              alt={`${title} source`}
+            />
+          )}
+          {!image && (
+            <div className="flex justify-center items-center -mt-14 w-24 h-24 bg-pepper-40 rounded-full z-10">
+              {icon}
+            </div>
+          )}
+          {membersCount > 0 && (
+            <SquadMemberShortList
+              squad={{
+                ...squad,
+                type: SourceType.Squad,
+              }}
+              members={members?.edges?.reduce((acc, current) => {
+                acc.push(current.node);
+                return acc;
+              }, [])}
+            />
+          )}
+        </div>
+        <div className="flex flex-col flex-1 justify-between">
+          <div className="flex-auto">
+            <div className="font-bold typo-title3">{title}</div>
+            <div className="mb-2 text-theme-label-secondary">{subtitle}</div>
+            <div className="mb-2 line-clamp-5 text-theme-label-secondary multi-truncate">
+              {description}
+            </div>
+          </div>
+
+          {!!action && type !== SourceType.Squad && (
+            <Button
+              className="w-full btn-secondary"
+              onClick={action?.onClick}
+              data-testid="source-action"
+            >
+              {action?.text}
+            </Button>
+          )}
+
+          {!!action && type === SourceType.Squad && (
+            <SquadJoinButton
+              className="w-full !btn-secondary"
+              squad={{
+                ...squad,
+                type,
+              }}
+              origin={Origin.SquadDirectory}
+              onSuccess={() => router.push(permalink)}
+              joinText={action?.text}
+              data-testid="squad-action"
+            />
+          )}
+
+          {!!link && (
+            <Button
+              className="w-full btn-secondary"
+              onClick={action?.onClick}
+              tag="a"
+              href={link.href}
+              target={link?.target ? link.target : '_self'}
```

Comment:
We have to make sure if it's external we add rel as well, there should be some examples around the codebase.